waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -240,15 +242,17 @@
   "Make an HTTP request to the Kubernetes API server using the configured authentication.
    If data is provided via :body, the application/json content type is added automatically.
    The response payload (if any) is automatically parsed to JSON."
-  [client url & {:keys [body content-type request-method] :as options}]
+  [client url {:keys [scheduler-name]} & {:keys [body content-type request-method] :as options}]
```

Comment:
In other places where I refactored to include the scheduler as a parameter, I replaced the `http-client` argument with the `scheduler` argument, since you can destructure the `scheduler` to get the `http-client`. I think that would be a good strategy to use here as well.